src/Views/TryInstallView.vala

```diff
@@ -16,99 +16,147 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-public class TryInstallView : AbstractInstallerView {
+public class Installer.TryInstallView : AbstractInstallerView {
+    public signal void custom_step ();
     public signal void next_step ();
 
-    construct {
-        var image = new Gtk.Image.from_icon_name ("system-os-installer", Gtk.IconSize.DIALOG);
-        image.valign = Gtk.Align.END;
-
-        var title_label = new Gtk.Label (_("Install or Try Demo Mode"));
-        title_label.max_width_chars = 60;
-        title_label.valign = Gtk.Align.START;
-        title_label.get_style_context ().add_class ("h2");
-
-        var choice_image = new Gtk.Image.from_icon_name ("computer-symbolic", Gtk.IconSize.LARGE_TOOLBAR);
-        choice_image.halign = Gtk.Align.END;
-        choice_image.valign = Gtk.Align.START;
-
-        var choice_label = new Gtk.Label (_("You can install %s on this device now, or try Demo Mode without installing.").printf (Utils.get_pretty_name ()));
-        choice_label.halign = Gtk.Align.START;
-        choice_label.max_width_chars = 52;
-        choice_label.valign = Gtk.Align.START;
-        choice_label.wrap = true;
-        choice_label.xalign = 0;
-
-        var implications_image = new Gtk.Image.from_icon_name ("document-revert-symbolic", Gtk.IconSize.LARGE_TOOLBAR);
-        implications_image.halign = Gtk.Align.END;
-        implications_image.valign = Gtk.Align.START;
-
-        var implications_label = new Gtk.Label (_("In Demo Mode, changes you make will not be saved and data from your previous operating system will be unchanged. Performance and features may not reflect the installed experience."));
-        implications_label.halign = Gtk.Align.START;
-        implications_label.max_width_chars = 52;
-        implications_label.valign = Gtk.Align.START;
-        implications_label.wrap = true;
-        implications_label.xalign = 0;
-
-        var return_image = new Gtk.Image.from_icon_name ("go-home-symbolic", Gtk.IconSize.LARGE_TOOLBAR);
-        return_image.halign = Gtk.Align.END;
-        return_image.valign = Gtk.Align.START;
-
-        var return_label = new Gtk.Label (_("You can always return to the installer from Demo Mode by selecting the Install icon."));
-        return_label.halign = Gtk.Align.START;
-        return_label.max_width_chars = 52;
-        return_label.valign = Gtk.Align.START;
-        return_label.wrap = true;
-        return_label.xalign = 0;
-
-        var grid = new Gtk.Grid ();
-        grid.column_spacing = 12;
-        grid.row_spacing = 32;
-
-        grid.attach (choice_image, 0, 1, 1, 1);
-        grid.attach (choice_label, 1, 1, 1, 1);
-        grid.attach (implications_image, 0, 2, 1, 1);
-        grid.attach (implications_label, 1, 2, 1, 1);
-        grid.attach (return_image, 0, 3, 1, 1);
-        grid.attach (return_label, 1, 3, 1, 1);
+    private Gtk.Button next_button;
+    private Gtk.Grid type_grid;
```

Comment:
This doesn't need to be declared here since it's only used inside the construct block. You can just do `var type_grid`